tests/integration.rs

```diff
@@ -6293,6 +6297,236 @@ mod common_parallel {
         handle_child_output(r, &output);
     }
 
+    #[test]
+    fn test_snapshot_restore_with_fd() {
+        let focal = UbuntuDiskConfig::new(FOCAL_IMAGE_NAME.to_string());
+        let guest = Guest::new(Box::new(focal));
+        let kernel_path = direct_kernel_boot_path();
+
+        let api_socket_source = format!("{}.1", temp_api_path(&guest.tmp_dir));
+
+        let net_id = "net123";
+        let num_queue_pairs: usize = 2;
+        // use a name that does not conflict with tap dev created from other tests
+        let tap_name = "chtap999";
+        use std::str::FromStr;
+        let taps = net_util::open_tap(
+            Some(tap_name),
+            Some(std::net::Ipv4Addr::from_str(&guest.network.host_ip).unwrap()),
+            None,
+            &mut None,
+            None,
+            num_queue_pairs,
+            Some(libc::O_RDWR | libc::O_NONBLOCK),
+        )
+        .unwrap();
+        let net_params = format!(
+            "id={},fd=[{},{}],mac={},ip={},mask=255.255.255.0,num_queues={}",
+            net_id,
+            taps[0].as_raw_fd(),
+            taps[1].as_raw_fd(),
+            guest.network.guest_mac,
+            guest.network.host_ip,
+            num_queue_pairs * 2
+        );
+
+        let mem_params = "size=2G";
+
+        let cloudinit_params = format!(
+            "path={},iommu=on",
+            guest.disk_config.disk(DiskType::CloudInit).unwrap()
+        );
+
+        let socket = temp_vsock_path(&guest.tmp_dir);
+        let event_path = temp_event_monitor_path(&guest.tmp_dir);
+
+        let mut child = GuestCommand::new(&guest)
+            .args(["--api-socket", &api_socket_source])
+            .args(["--event-monitor", format!("path={event_path}").as_str()])
+            .args(["--cpus", "boot=4"])
+            .args(["--memory", mem_params])
+            .args(["--balloon", "size=0"])
+            .args(["--kernel", kernel_path.to_str().unwrap()])
+            .args([
+                "--disk",
+                format!(
+                    "path={}",
+                    guest.disk_config.disk(DiskType::OperatingSystem).unwrap()
+                )
+                .as_str(),
+                cloudinit_params.as_str(),
+            ])
+            .args(["--net", net_params.as_str()])
+            .args(["--vsock", format!("cid=3,socket={socket}").as_str()])
```

Comment:
No need for vsock.

Reply:
Thanks for these suggestions @rbradford. Would you recommend making these changes on other existing snapshot tests as well (i.e., remove vsock, reduce memory and vcpu) ?

Reply:
Okay, please move this and other snapshot/restore tests to `sequential_common`.

Reply:
With vsock removal changes on just the new test added, I trigged the PR workflow few times to observe the stability overall. While it reduces the number of instances of errors, I could still see the failures. ARM tests job failed more than x86-64 tests.
So, I went ahead and moved all the snapshot/restore tests to common_sequential. But git does not show the diff as "code chunk removed + code chunk added", it would mess up the diff and blame, see https://github.com/cloud-hypervisor/cloud-hypervisor/commit/9ddd3d67622575caf70ce3ed6d7b8075fc66a333
So, I had to go about a slightly ugly way of splitting the tests move into 2 commits - 
1. remove tests from commom_parallel 
2. add tests to common_sequential

Please let me know if you have any other suggestions on this, @rbradford